drivers/serial/serial.c

```diff
@@ -485,6 +489,87 @@ static int uart_tcdrain(FAR uart_dev_t *dev,
   return ret;
 }
 
+/****************************************************************************
+ * Name: uart_tcsendbreak
+ *
+ * Description:
+ *   Request a serial line Break by calling the lower half driver's
+ *   BSD-compatible Break IOCTLs TIOCSBRK and TIOCCBRK, with a sleep of the
+ *   specified duration between them.
+ *
+ * Input Parameters:
+ *   dev      - Serial device.
+ *   filep    - Required for issuing lower half driver IOCTL call.
+ *   ms       - If non-zero, duration of the Break in milliseconds; if
+ *              zero, duration is 400 milliseconds.
+ *
+ * Returned Value:
+ *   0 on success or a negated errno value on failure.
+ *
+ ****************************************************************************/
+
+static int uart_tcsendbreak(FAR uart_dev_t *dev, FAR struct file *filep,
+                            unsigned int ms)
+{
+  int ret;
+
+  /* tcsendbreak is a cancellation point */
+
+  if (enter_cancellation_point())
+    {
+#ifdef CONFIG_CANCELLATION_POINTS
```

Comment:
should we move before line 518?

Reply:
I don't know; in other functions (e.g., see uart_tcdrain() above this function) `#ifdef CONFIG_CANCELLATION_POINTS` is after `if (enter_cancellation_point())`. If order should be changed, then perhaps we are not consistent everywhere?

Reply:
ok, let keep it mow.